pkg/services/ngalert/state/historian/annotation_test.go

```diff
@@ -49,7 +51,8 @@ func createTestAnnotationBackendSut(t *testing.T) *AnnotationBackend {
 	rules.Rules[1] = []*models.AlertRule{
 		models.AlertRuleGen(withOrgID(1), withUID("my-rule"))(),
 	}
-	return NewAnnotationBackend(fakeAnnoRepo, &dashboards.FakeDashboardService{}, rules)
+	metrics := metrics.NewHistorianMetrics(prometheus.NewRegistry())
+	return NewAnnotationBackend(fakeAnnoRepo, &dashboards.FakeDashboardService{}, rules, metrics)
```

Comment:
If possible, I'd like you test the metrics themselves - Prometheus has utils to be able to do that, I'm happy to suggest something if you can point me to which file you'd do the test on (or if you'd rather do one per backend e.g. one test in `annotation_test.go` and one in `loki_http_test.go`

Reply:
Lots of tests added - there is one test for all metrics in each of `annotation_test.go` and `loki_test.go`. There are other tests added as well, infrastructure type stuff, the two metrics tests are the main focus.